### PR TITLE
Initial update of Module.md for WebSub extension

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
Initial update of Module.md for WebSub extension.

TODO: Need to add a detailed example.

### CODE REVIEW

File: stdlib/websub/src/main/ballerina/websub/Module.md [hunk 1]
```diff
@@ -255,3 +255,99 @@ where the values specified via the Config API would override values specified as
...
+        "issue" : {
+            "<PAYLOAD_KEY_TO_CONSIDER>" : {
+                "opened" : ("onIssueOpened", IssueOpenedEvent),
+                "closed" : ("onIssueClosed", IssueClosedEvent)
+            }
+        }
+    }
+};
+```
+`"opened" : ("onIssueOpened", IssueOpenedEvent)` here indicates that when the value of the header to consider
```
Comment (samgnaniah) [inline stdlib/websub/src/main/ballerina/websub/Module.md]: I found this sentence really hard to follow. Perhaps it's just too long so can we consider shortening it to make it simpler?

File: stdlib/websub/src/main/ballerina/websub/Module.md [hunk 2]
```diff
@@ -255,3 +255,99 @@ where the values specified via the Config API would override values specified as
 | b7a.websub.hub.remotepublish   | false         | Whether publishing updates against the topics in the hub could be done by remote publishers via HTTP requests with `hub.mode` set to `publish`  |
 | b7a.websub.hub.topicregistration | true      | Whether a topic needs to be registered at the hub for publishers to publish updates against the topic and for subscribers to send subscription requests for the topic |
 | b7a.websub.hub.enablessl | true      | Whether the Hub service should be exposed over HTTPS |
+
+## Introducing Specific Subscriber Services
+
+Ballerina's WebSub subscriber service endpoint can be extended to introduce specific Webhooks.
+ 
+With specific subscriber services, multiple resources corresponding to the possible notifications for a particular
```
Comment (samgnaniah) [inline stdlib/websub/src/main/ballerina/websub/Module.md]: Passive voice doesn't work so well in this case as it makes it harder to understand. Something like this with the subject first.

"With specific subscriber services, you can introduce multiple resources corresponding to the possible notifications for a particular topic, instead of the single `onNotification` resource."

Alternatively, you could start the sentence with "You". 

In general use active voice more where the subject (in this case "You" comes first).

File: stdlib/websub/src/main/ballerina/websub/Module.md [hunk 3]
```diff
@@ -255,3 +255,99 @@ where the values specified via the Config API would override values specified as
...
+    topicIdentifier: websub:TOPIC_ID_HEADER,
+    topicHeader: "<HEADER_TO_CONSIDER>",
+    headerResourceMap: {
+        "issueOpened" : ("onIssueOpened", IssueOpenedEvent),
+        "issueClosed" : ("onIssueClosed", IssueClosedEvent)
+    }
+};
+```
+
+`"issueOpened" : ("onIssueOpened", IssueOpenedEvent)` here indicates that when the value of the header to consider is
```
Comment (samgnaniah) [inline stdlib/websub/src/main/ballerina/websub/Module.md]: This sentence makes sense but it's too long. People might struggle to understand.

File: stdlib/websub/src/main/ballerina/websub/Module.md [hunk 4]
```diff
@@ -255,3 +255,99 @@ where the values specified via the Config API would override values specified as
...
+    payloadKeyResourceMap: {
+        "<PAYLOAD_KEY_TO_CONSIDER>" : {
+            "issueOpened" : ("onIssueOpened", IssueOpenedEvent),
+            "issueClosed" : ("onIssueClosed", IssueClosedEvent)
+        }
+    }
+};
+```
+
+`"issueOpened" : ("onIssueOpened", IssueOpenedEvent)` here indicates that when the value of the JSON payload key to
```
Comment (samgnaniah) [inline stdlib/websub/src/main/ballerina/websub/Module.md]: Same issue regarding the length of the sentence.

File: stdlib/websub/src/main/ballerina/websub/Module.md [hunk 5]
```diff
@@ -255,3 +255,99 @@ where the values specified via the Config API would override values specified as
...
+
+`"issueOpened" : ("onIssueOpened", IssueOpenedEvent)` here indicates that when the value of the JSON payload key to 
+consider is `issueOpened`, dispatching needs to happen to a resource named `onIssueOpened` whose 
+ first parameter is the generic `websub:Notification` record, and the second parameter is the custom 
+ `IssueOpenedEvent` record mapping the JSON payload received when an issue is created.  
+
+**Based on a request header and the payload**
+ 
+Dispatching would be based on both a request header and the payload as specified in `headerAndPayloadKeyResourceMap`. 
+Additionally one may also introduce a `headerResourceMap` and/or `payloadKeyResourceMap` as additional mappings.
```
Comment (samgnaniah) [inline stdlib/websub/src/main/ballerina/websub/Module.md]: Change to "Additionally, you may..".

File: stdlib/websub/src/main/ballerina/websub/Module.md (hunk 2 above)
Comment (MaryamZi) [reply stdlib/websub/src/main/ballerina/websub/Module.md]: IINM, it seems we haven't used active voice in the Module.md files. I've updated the content here, but still not active voice since it might seem inconsistent, WDYT?

**Comment by MaryamZi** [comment]
Closing this PR temporarily. Will re-open with a concrete example introduced here and at https://ballerina.io/learn/how-to-extend-ballerina/ (to link to).

File: stdlib/websub/src/main/ballerina/websub/Module.md (hunk 2 above)
Comment (samgnaniah) [reply stdlib/websub/src/main/ballerina/websub/Module.md]: Yes, you're right. Module.md files are mostly in passive voice. We used that approach because we wanted to be more direct and less conversational there. I think you can use passive in this case as well. My bad.
